noether_filtering/include/noether_filtering/impl/filter_group.hpp

```diff
@@ -200,13 +200,19 @@ bool FilterGroup<F>::applyFilters(const std::vector<std::string>& filters, const
     {
       err_msg = boost::str(boost::format("The filter %s failed") % fname);
       CONSOLE_BRIDGE_logError("%s", err_msg.c_str());
-    }
 
-    if(!current_filter_succeeded && !continue_on_failure_)
+      if(!continue_on_failure_)
+      {
+        return false;
+      }
+
+    }
+    else
     {
-      return false;
+      // save output onto the input for the next filter
+      temp = std::move(temp_output);
     }
-    temp = std::move(temp_output);
+
     success |= current_filter_succeeded;
```

Comment:
Not sure how we missed this but this logic is incorrect. Imagine the scenario where the first filter succeeds but the second filter fails:
1. success = false
1. success = false | false = false
1. success = false | true = true

Even though one of the filters failed, success would still be true which is wrong. We should start with `success = true` and update that value by `success &= current_filter_succeeded`.
1. success = true
1. success = true & false = false
1. success = false & true = false

We should revisit the return value of this function. What does it mean to return a variable indicating success if someone sets `continue_on_failure = true`? Maybe we don't even need to keep track of success at all, just return false as implemented above and return true otherwise.

Reply:
When `continue_on_failure_ = true` is set to true then it only takes one filter to succeed for the whole pass through the group to succeed.  When `continue_on_failure_ = false`  then the algorithm returns false as soon as one filter fails.  I think that logic is enforced correctly here

Reply:
I see. I was under the impression that the return of the function was an indication of the success of the entire pipeline, regardless of whether or not you wanted it to continue on failure.  In that view, a pipeline where a single element fails (despite whether you continue on failure or not) would make this function return false. In the context of this PR, I am okay with the existing implementation, but we should document your comment above in the code before merging.

I think this raises a bigger question about what it means to continue after failure in a serial pipeline. To me, it doesn't really seem conceptually valid and I can't immediately think of a use-case where this functionality would be helpful/required. However, that discussion is outside the scope of this PR, so I'll open a separate issue